tests/lib/rules/prefer-object-has-own.js

```diff
@@ -0,0 +1,263 @@
+/**
+ * @fileoverview Prefers Object.hasOwn instead of Object.prototype.hasOwnProperty
+ * @author Nitin Kumar
+ * @author Gautam Arora
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require("../../../lib/rules/prefer-object-has-own");
+const { RuleTester } = require("../../../lib/rule-tester");
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const parserOptions = {
+    ecmaVersion: 2022
+};
+
+const ruleTester = new RuleTester({ parserOptions });
+
+ruleTester.run("prefer-object-has-own", rule, {
+    valid: [
```

Comment:
Valid test case: `hasOwnProperty(obj, prop)`

Reply:
Added in https://github.com/eslint/eslint/pull/15346/commits/6c279152f5009f02f4af5c7fca1d100b2634c2a5